api/base/utils.py

```diff
@@ -232,14 +232,16 @@ class MockQueryset(list):
     This class is meant to convert a simple list into a filterable queryset look-a-like.
     """
 
-    def __len__(self, *args, **kwargs):
-        if kwargs.get('total'):
-            self.total = kwargs.get('total')
+    def __init__(self, items, search, default_attrs=None, **kwargs):
```

Comment:
this `MockQueryset` is kinda diverging from its stated purpose (`list` that looks like a queryset) -- it's now more specific to this particular use case than a utility should be.

not a blocky problem, since it's only used in the one place and is unlikely to be used in other situations, but those are also reasons it probably shouldn't be in `utils`